client/src/components/map/controls/map-style.tsx

```diff
@@ -0,0 +1,38 @@
+import { RadioGroup, RadioGroupIndicator, RadioGroupItem } from "@/components/ui/radio-group";
+import Image from "next/image";
+import { mapStyleOptions } from "../constants";
+import { MapStyles } from "../types";
+
+export type MapStyleOptionsProps = {
+  onChangeMapStyle: (style: MapStyles) => void;
+  defaultValue: MapStyles;
+};
+
+const MapStyleControl = ({ onChangeMapStyle, defaultValue }: MapStyleOptionsProps) => {
+  return (
+    <RadioGroup onValueChange={onChangeMapStyle} defaultValue={defaultValue} className="space-y-2">
+      {mapStyleOptions.map((option) => (
+        <div key={option.value} className="flex items-center gap-2">
+          <RadioGroupItem
+            value={option.value}
+            id={option.value}
+            className="flex items-center justify-center border-none"
+          >
+            <Image src={option.icon} alt={option.label} width={20} height={20} />
+            <RadioGroupIndicator asChild>
+              <div className="absolute h-[26px] w-[26px] rounded-full border border-global" />
+            </RadioGroupIndicator>
+          </RadioGroupItem>
+          <label
```

Comment:
There's a specific style for the label when the button has focus: https://www.figma.com/design/HBpMppIFHR1vA0SbnUJGfO/GRASS---Design-%5BInternal%5D?node-id=536-3375&t=986E6GmpDtbfvv5L-4